src/main/java/net/tridentsdk/chat/ChatComponent.java

```diff
@@ -0,0 +1,482 @@
+            }
+        }
+        $lock.readLock().unlock();
+        return false;
+    }
+
+    /**
+     * Gets whether or not this component is bold.
+     *
+     * @return True iff it is.
+     */
+    public boolean isBold() {
+        return bold != null && bold.get();
+    }
+
+    /**
+     * Sets this component's boldness.
+     *
+     * @param bold The boldness.
+     *
+     * @return This component.
+     */
+    public ChatComponent setBold(boolean bold) {
+        $lock.writeLock().lock();
+        if (this.bold == null)
+            this.bold = new AtomicBoolean(bold);
+        else
+            this.bold.set(bold);
+        $lock.writeLock().unlock();
+        return this;
+    }
+
+    /**
+     * Gets whether or not this component is in italics.
+     *
+     * @return True iff it is.
+     */
+    public boolean isItalic() {
+        return italic != null && italic.get();
+    }
+
+    /**
+     * Sets this component's italic.
+     *
+     * @param italic The italic.
+     *
+     * @return This component.
+     */
+    public ChatComponent setItalic(boolean italic) {
+        $lock.writeLock().lock();
+        if (this.italic == null)
+            this.italic = new AtomicBoolean(italic);
+        else
+            this.italic.set(italic);
+        $lock.writeLock().unlock();
+        return this;
+    }
+
+    /**
+     * Gets whether or not this component is underlined.
+     *
+     * @return True iff it is.
+     */
+    public boolean isUnderlined() {
+        return underlined != null && underlined.get();
+    }
+
+    /**
+     * Sets this component's underlined.
+     *
+     * @param underlined The underline.
+     *
+     * @return This component.
+     */
+    public ChatComponent setUnderlined(boolean underlined) {
+        $lock.writeLock().lock();
+        if (this.underlined == null)
+            this.underlined = new AtomicBoolean(underlined);
+        else
+            this.underlined.set(underlined);
+        $lock.writeLock().unlock();
+        return this;
+    }
+
+    /**
+     * Gets whether or not this component is striked through.
+     *
+     * @return True iff it is.
+     */
+    public boolean isStrikethrough() {
+        return strikethrough != null && strikethrough.get();
+    }
+
+    /**
+     * Sets this component's strikethrough.
+     *
+     * @param strikethrough The strikethrough.
+     *
+     * @return This component.
+     */
+    public ChatComponent setStrikethrough(boolean strikethrough) {
+        $lock.writeLock().lock();
+        if (this.strikethrough == null)
+            this.strikethrough = new AtomicBoolean(strikethrough);
+        else
+            this.strikethrough.set(strikethrough);
+        $lock.writeLock().unlock();
+        return this;
+    }
+
+    /**
+     * Gets whether or not this component is in obfuscation.
+     *
+     * @return True iff it is.
+     */
+    public boolean isObfuscated() {
+        return obfuscated != null && obfuscated.get();
```

Comment:
THREAD-SAFETY
This is a poor construct to use for a thread-safe class. It doesn't save (that much) memory, and can be solved by using a volatile boolean set to false as a default value instead.